plan/plan.go

```diff
@@ -274,12 +288,29 @@ func (p *Plan) buildAccounts(c *v2.Config) map[string]Account {
 
 		accountPlan.AccountName = name
 		accountPlan.ComponentCommon = resolveComponentCommon(defaults.Common, acct.Common)
-		accountPlan.AllAccounts = resolveAccounts(c.Accounts)
+		if accountPlan.ComponentCommon.Backend.Kind == BackendKindS3 {
+			accountPlan.ComponentCommon.Backend.S3.KeyPath = fmt.Sprintf("terraform/%s/accounts/%s.tfstate", accountPlan.ComponentCommon.Project, name)
+		} else if accountPlan.ComponentCommon.Backend.Kind == BackendKindRemote {
+			accountPlan.ComponentCommon.Backend.Remote.Workspace = fmt.Sprintf("accounts-%s", name)
+		}
```

Comment:
should the `else` be an assertion/panic? in case we introduce a bug or later decide to add another one

Reply:
Yeah. I was working under the belief that we are protected by the type checker, but had not thought about the future-proofing angle.